contracts/modules/cover/Cover.sol

```diff
@@ -501,6 +501,10 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     string calldata ipfsDescriptionHash
   ) external whenNotPaused returns (uint /*poolId*/, address /*stakingPoolAddress*/) {
 
+    uint numProducts = productInitParams.length;
+
+    (uint poolId, address stakingPoolAddress) = stakingPoolFactory.create(address(this));
```

Comment:
was this moved here just to have access to the poolId number?

Could also be predicted because it's always +1

You want to run validation first before doing the actual updates (gas reasons, security reasons), so i'd say let's see if we can keep this after validation